knative/eventing
# Populate a Subscriptions subscriber and reply namespace only if not set already

Fixes #6539

In case a Subscription with a `.spec.subscriber.ref` pointing to a service in another namespace, the Subscription will not become ready because it is trying to find the service on the subscription's namespace (see #6539). Same happens with a subscriptions `.spec.reply.ref`. 
This PR addresses it and only uses the namespace of the Subscription, if no namespace was given for the subscriber/reply.

## Proposed Changes
* :bug: Populate a subscriptions subscriber & reply namespace field with the Subscriptions namespace only in case it is not set

### Pre-review Checklist
- [ ] **At least 80% unit test coverage**
- [ ] **E2E tests** for any new behavior
- [ ] **Docs PR** for any user-facing impact
- [ ] **Spec PR** for any new API feature
- [ ] **Conformance test** for any change to the spec

**Release Note**
```release-note

```

**How to verify**
1. Install a eventing with this patch
2. Create a sequence with steps and a reply in different namespaces (e.g. https://gist.github.com/creydr/57235f35b03ab8f1285eee0a64350576)
3. Check that the subscriptions and sequence becomes ready
